deepface/detectors/FastMtCnn.py

```diff
@@ -68,7 +69,9 @@ def build_model(self) -> Any:
                 "Please install using 'pip install facenet-pytorch' "
             ) from e
 
-        face_detector = fast_mtcnn(device="cpu")
+        device = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')
+        face_detector = fast_mtcnn(device=device)
```

Comment:
to import fast_mtcnn, we [enforce](https://github.com/serengil/deepface/blob/master/deepface/detectors/FastMtCnn.py#L64) users to install facenet_pytorch instead of torch. not sure installing facenet_pytorch will install torch, too.

Reply:
okay I can confirm that installing facenet-pytorch will install torchvision, and installing torchvision will install torch

```shell
(base) sefik@L-GCCDXDDT:~$ pip show facenet-pytorch
Name: facenet-pytorch
Version: 2.5.3
Summary: Pretrained Pytorch face detection and recognition models
Home-page: https://github.com/timesler/facenet-pytorch
Author: Tim Esler
Author-email: tim.esler@gmail.com
License:
Location: /home/sefik/miniconda3/lib/python3.9/site-packages
Requires: numpy, pillow, requests, torchvision
Required-by:


(base) sefik@L-GCCDXDDT:~$ pip show torchvision
Name: torchvision
Version: 0.15.2
Summary: image and video datasets and models for torch deep learning
Home-page: https://github.com/pytorch/vision
Author: PyTorch Core Team
Author-email: soumith@pytorch.org
License: BSD
Location: /home/sefik/miniconda3/lib/python3.9/site-packages
Requires: numpy, pillow, requests, torch
Required-by: facenet-pytorch, timm, ultralytics
```

But still putting `import torch` to the line 5 will throw exception for users who do not have facenet-pytorch. Because when i run `pip install deepface`, it will not install torch (it is not a must dependency) and when i import deepface as `from deepface import DeepFace`, then torch will be attempted to be imported but it will throw exception because I do not have that dependency. So, this change breaks deepface code as is.

As you can see facenet_pytorch is imported in line 64 instead of global level: https://github.com/serengil/deepface/blob/master/deepface/detectors/FastMtCnn.py#L64

You may consider to import torch in that try except block.